src/nbt/compound.rs

```diff
@@ -50,6 +50,32 @@ impl NbtCompound {
         bytes.freeze()
     }
 
+    pub fn from_values(values: Vec<(&str, NbtTag)>) -> Self {
+        let mut child_tags = HashMap::new();
+        for (name, tag) in values {
+            let name_string = name.to_string();
+            child_tags.insert(name_string, tag);
+        }
+        Self { child_tags }
+    }
+
+    /// Puts a value into the compound
+    /// # Examples
+    /// ```
+    /// # use crab_nbt::NbtCompound;
+    /// let mut compound = NbtCompound::default();
+    /// compound.put("string".to_owned(), "string");
+    /// compound.put("bool".to_owned(), true);
+    /// compound.put("int".to_owned(), 10);
+    ///
+    /// assert_eq!(compound.get_string("string"), Some(&"string".to_owned()));
+    /// assert_eq!(compound.get_bool("bool"), Some(true));
+    /// assert_eq!(compound.get_int("int"), Some(10));
+    /// ```
+    pub fn put(&mut self, name: String, value: impl Into<NbtTag>) {
```

Comment:
why not name: &str?

Reply:
Because then your need to convert it to `String`, and rust api guidlines recommend to take the owned type in such case.